great_expectations/data_context/types/base.py

```diff
@@ -1791,6 +1792,13 @@ def update(
 
                 other_batch_request = other_config.batch_request
 
+                # Necessary when using RuntimeDataConnector with SimpleCheckpoint
```

Comment:
@cdkini I agree with the logic of this change.  It seems to me that its place should be in the caller.  Based on my experience and understanding, all arguments to the constructors of "config" objects (such as `CheckpointConfig`) in `base.py` are already expected to be basic types (i.e., not typed objects, such as `BatchRequest` types).